src/debugger/channel/MessageHandler.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+/* @flow */
+import { DebugMessage } from "./DebugMessage.js";
+import type { BreakpointArguments } from "./../types.js";
+import invariant from "./../../invariant.js";
+
+export class MessageHandler {
```

Comment:
We are actually designing a small RPC system right now. In RPC terminology, we normally call this "MessageMarshaller", "formatXXX" => "marshalXXX", "parseXXX" => "unmarshalXXX".